src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/INuGetSourceRepositoryService.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Configuration;
+using NuGet.PackageManagement.VisualStudio.RemoteTypes;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.PackageManagement.VisualStudio
+{
+    public interface INuGetSourceRepositoryService : IDisposable
```

Comment:
This interface might be a bit too rich right now. 
We should try to keep the contracts in a contracts assembly (like NuGet.VisualStudio.Internal.Contracts)

Reply:
Consider how this evolves over time. If you add properties, will you need to add members to this RPC interface? How chatty is this going to be when initializing the UI? Will it require many round-trips (over potentially high-latency networks)?

As an alternative, consider defining a class with members for each of these properties, including collections, etc. Then just have a few methods on the RPC interface that get the whole instance of a class at once, or store changes made to it all at once. You may be able to add members to that class more easily than adding members to an interface over time.